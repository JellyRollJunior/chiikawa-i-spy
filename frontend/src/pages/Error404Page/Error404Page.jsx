import { Header } from '../../components/Header/Header.jsx';
import { ErrorElement } from '../../components/ErrorElement/ErrorElement.jsx';
import { Link } from 'react-router';
import { Footer } from '../../components/Footer/Footer.jsx';
import shared from '../../styles/shared.module.css';

const Error404Page = () => {
  return (
    <>
      <div>
        <Header />
        <main className={shared.marginTopMedium}>
          <ErrorElement error={'404: Page not found.'} msgId={1}>
            <Link to="/" className={shared.marginTopSmall}>
              <button className={shared.primaryButton}>
                Return to homepage
              </button>
            </Link>
          </ErrorElement>
        </main>
      </div>
      <Footer />
    </>
  );
};

export { Error404Page };
