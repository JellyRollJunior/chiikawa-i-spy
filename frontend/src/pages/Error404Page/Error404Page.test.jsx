import { vi, describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { Error404Page } from './Error404Page.jsx';
import { BrowserRouter } from 'react-router';

vi.mock('../../components/Header/Header.jsx', () => ({
  Header: () => <header>I am a header</header>,
}));

vi.mock('../../components/Footer/Footer.jsx', () => ({
  Footer: () => <footer>I am a footer</footer>,
}));

vi.mock('../../components/IconWrapper/IconWrapper.jsx', () => ({
  IconWrapper: ({ children }) => {
    children;
  },
}));

describe('Error 404 page', () => {
  it('renders header, footer, error message, return home button', () => {
    const { container } = render(
      <BrowserRouter>
        <Error404Page />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
