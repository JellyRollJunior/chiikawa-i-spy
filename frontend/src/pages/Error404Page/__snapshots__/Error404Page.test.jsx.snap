// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error 404 page > renders header, footer, error message, return home button 1`] = `
<div>
  <div>
    <header>
      I am a header
    </header>
    <main
      class="_marginTopMedium_5890fc"
    >
      <section
        class="_card_5890fc _errorWrapper_ef92cb _marginTopMedium_5890fc"
      >
        <img
          alt="Ancient wise rabbit"
          class="_errorUsagi_ef92cb _marginTopSmall_5890fc"
          src="/src/assets/error/usagi.png"
        />
        <p>
          Ancient wise rabbit
           once said:
          <br />
          Yaha Yaha unma unma Yaha
        </p>
        <a
          class="_marginTopSmall_5890fc"
          data-discover="true"
          href="/"
        >
          <button
            class="_primaryButton_5890fc"
          >
            Return to homepage
          </button>
        </a>
      </section>
    </main>
  </div>
  <footer>
    I am a footer
  </footer>
</div>
`;
